docs/public/components/shared_features/flares.md

```diff
@@ -0,0 +1,119 @@
+# Flare
+
+The general idea is to register a callback within your component to be called each time a flare is created. This uses
+[Fx](../fx.md) groups under the hood, but helpers are there to abstract all the complexity.
+
+Once the callback is created, you will have to migrate the code related to your component from `pkg/flare` to your
+component.
+
+## Creating a callback
+
+To add data to a flare, you first need to register a callback, also known as a `FlareBuilder`.
+
+Within your component, create a method with the following signature: `func (c *yourComp) fillFlare(fb flaretypes.FlareBuilder) error`.
+
+This function is called every time the Agent generates a flare—whether from the CLI, RemoteConfig, or from the running
+Agent. Your callback takes a
+[FlareBuilder](https://pkg.go.dev/github.com/DataDog/datadog-agent/comp/core/flare/types#FlareBuilder) as parameter.
+This object provides all the helpers functions needed to add data to a flare (adding files, copying
+directories, scrubbing data, and so on).
+
+Example:
+
+```golang
+import (
+	yaml "gopkg.in/yaml.v2"
+
+	flaretypes "github.com/DataDog/datadog-agent/comp/core/flare/types"
```

Comment:
I have a PR up currently that splits this up into two folders: `.../flare/def` and `.../flare/types`. The former is usable without fx, while the latter is kept for backwards compatibility. So, it may be worth using `flare/def` here in order to stay more general; the usage of each is identical and the types are aliases so they stay compatible.